python/lib/modeldata/dmod/modeldata/data/object_store_manager.py

```diff
@@ -249,8 +250,15 @@ def add_data(self, dataset_name: str, dest: str, data: Optional[bytes] = None, s
                                                       retain_until_date=datetime.now() + timedelta(hours=1))
             else:
                 retention = None
-            result = self._client.put_object(bucket_name=dataset_name, data=io.BytesIO(data), length=len(data),
-                                             object_name=dest, retention=retention)
+            if isinstance(data, Reader):
+                # Use AWS S3 default part size of 5MiB
+                # https://docs.aws.amazon.com/AmazonS3/latest/userguide/qfacts.html
+                part_size = 5 * 1024 * 1024
```

Comment:
Might it make sense to have this be a defined constant within the _Reader_ interface?

Reply:
I don't think so. The concept of a `Reader` does not need to know about part size IMO. Part size, from my understanding, is used by minio and S3 to chunk up an upload. If we run into performance issues here with uploading larger files in the future, we can either accept a different variant of a `Reader` that also implements `__len__` so we can more appropriately select a `part_size`, _or_ just take in an optional `part_size` in the method signature. At that point, we might just want to create a different chunked upload method to handle that case.